lang-dotnet/mono/autobuild/patches/0002-LoongArch64-migrate-to-new-world.patch

```diff
@@ -0,0 +1,150 @@
+--- a/mono/mini/mini-arch.h
++++ b/mono/mini/mini-arch.h
+@@ -23,7 +23,7 @@
+ #include "mini-arm.h"
+ #elif defined(TARGET_ARM64)
+ #include "mini-arm64.h"
+-#elif defined(__loongarch64)
++#elif defined(__loongarch__)
+ #include "mini-loongarch64.h"
+ #elif defined(__mips__)
+ #include "mini-mips.h"
+diff --git a/mono/sgen/sgen-archdep.h b/mono/sgen/sgen-archdep.h
+index 7768200e377..f9df0fdee6b 100644
+--- a/mono/sgen/sgen-archdep.h
++++ b/mono/sgen/sgen-archdep.h
+@@ -71,7 +71,7 @@
+ 
+ #define REDZONE_SIZE	0
+ 
+-#elif defined(__loongarch64)
++#elif defined(__loongarch__)
+ 
+ #define REDZONE_SIZE	0
+ 
+diff --git a/mono/utils/mono-context.c b/mono/utils/mono-context.c
+index 38fb964c034..2568ed1775a 100644
+--- a/mono/utils/mono-context.c
++++ b/mono/utils/mono-context.c
+@@ -667,7 +667,7 @@ mono_monoctx_to_sigctx (MonoContext *mctx, void *sigctx)
+ #endif
+ }
+ 
+-#elif (defined(__loongarch64) && !defined(MONO_CROSS_COMPILE)) || (defined(TARGET_LOONGARCH64))
++#elif (defined(__loongarch__) && !defined(MONO_CROSS_COMPILE)) || (defined(TARGET_LOONGARCH64))
+ 
+ #include <mono/utils/mono-context.h>
+ #include <mono/utils/ftnptr.h>
+@@ -681,9 +681,9 @@ mono_sigctx_to_monoctx (void *sigctx, MonoContext *mctx)
+ 	memcpy (mctx->regs, UCONTEXT_GREGS (sigctx), sizeof (host_mgreg_t) * 32);
+ 	mctx->pc = UCONTEXT_REG_PC (sigctx);
+ 	mctx->regs [loongarch_sp] = UCONTEXT_REG_SP (sigctx);
+-	union __loongarch_mc_fp_state *fregs = (union __loongarch_mc_fp_state *)UCONTEXT_FPREGS (sigctx);
++	double *src_fregs = UCONTEXT_FPREGS (sigctx);
+ 	for (int i = 0; i < 32; i++) {
+-		mctx->fregs [i] = fregs [i].__val64 [0];
++		mctx->fregs [i] = src_fregs [i];
+ 	}
+ #endif
+ }
+diff --git a/mono/utils/mono-context.h b/mono/utils/mono-context.h
+index 8e423acb398..1e7b62ffffe 100644
+--- a/mono/utils/mono-context.h
++++ b/mono/utils/mono-context.h
+@@ -1058,7 +1058,7 @@ typedef struct {
+ 
+ #define MONO_ARCH_HAS_MONO_CONTEXT (1)
+ 
+-#elif (defined(__loongarch64) && !defined(MONO_CROSS_COMPILE)) || (defined(TARGET_LOONGARCH64))
++#elif (defined(__loongarch__) && !defined(MONO_CROSS_COMPILE)) || (defined(TARGET_LOONGARCH64))
+ 
+ #include <mono/arch/loongarch64/loongarch64-codegen.h>
+ 
+diff --git a/mono/utils/mono-sigcontext.h b/mono/utils/mono-sigcontext.h
+index 2153c1cad57..767134b7541 100644
+--- a/mono/utils/mono-sigcontext.h
++++ b/mono/utils/mono-sigcontext.h
+@@ -563,7 +563,7 @@ typedef struct ucontext
+ # define UCONTEXT_REG_Rn(ctx, n) (((ucontext_t *)(ctx))->uc_mcontext.gregs[(n)])
+ # define UCONTEXT_IP(ctx)         (((ucontext_t *)(ctx))->uc_mcontext.psw.addr)
+ 
+-#elif defined(__loongarch64)
++#elif defined(__loongarch__)
+ 
+ # if HAVE_UCONTEXT_H
+ #  include <ucontext.h>
+@@ -578,7 +578,7 @@ typedef struct ucontext
+ 	 } while (0)
+ #endif
+ 
+-# define UCONTEXT_FPREGS(ctx)	((double*)(((ucontext_t *)(ctx))->uc_mcontext.__fpregs))
++# define UCONTEXT_FPREGS(ctx)	((double*)(((ucontext_t *)(ctx))->uc_mcontext.__extcontext))
```

Comment:
In theory you should iterate over the linked list structure and find the one with matching magic. Please see [the UAPI `asm/sigcontext.h` file](https://github.com/torvalds/linux/blob/v6.16/arch/loongarch/include/uapi/asm/sigcontext.h) for details.